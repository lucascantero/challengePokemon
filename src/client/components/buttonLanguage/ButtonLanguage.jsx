import React from 'react';
import { useTranslation, withTranslation } from 'react-i18next';

const ButtonLanguage = () => {

    const { t, i18n } = useTranslation();

    const changeLanguage = ln => {
        return () => {
            i18n.changeLanguage(ln)
        }
    }

    return (
        <div className="buttonLanguage__wrapper">
            <button className="buttonLanguage__button" onClick = {changeLanguage('en')}>{t('en')}</button>
            <button className="buttonLanguage__button" onClick = {changeLanguage('es')}>{t('es')}</button>
        </div>
    ) 

}

export default withTranslation()(ButtonLanguage);