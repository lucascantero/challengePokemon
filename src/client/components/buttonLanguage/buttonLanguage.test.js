import React from 'react';
import ButtonLanguage from './ButtonLanguage.jsx';
import { act } from 'react-dom/test-utils';
import {render,unmountComponentAtNode} from 'react-dom';
import { I18nextProvider } from 'react-i18next';
import i18next from '../../i18n/index';

describe('Testing ButtonLanguage', () => {

    let container = null;

    beforeEach(() => {
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        unmountComponentAtNode(container);
        container.remove();
        container = null;
    });

    it('Should match snapshot', () => {



        act(() => {
            render( <I18nextProvider i18n={i18next}> <ButtonLanguage /> </I18nextProvider>, container)
        })

        expect(container.innerHTML).toMatchSnapshot();
    })

})