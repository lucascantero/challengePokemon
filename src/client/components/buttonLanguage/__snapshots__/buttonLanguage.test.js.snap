// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ButtonLanguage Should match snapshot 1`] = `" <div class=\\"buttonLanguage__wrapper\\"><button class=\\"buttonLanguage__button\\">Inglés</button><button class=\\"buttonLanguage__button\\">Español</button></div> "`;
